python/sglang/srt/managers/scheduler.py

```diff
@@ -2227,23 +2225,42 @@ def get_weights_by_name(self, recv_req: GetWeightsByNameReqInput):
         return GetWeightsByNameReqOutput(parameter)
 
     def release_memory_occupation(self, recv_req: ReleaseMemoryOccupationReqInput):
-        self.memory_saver_adapter.check_validity(
-            caller_name="release_memory_occupation"
-        )
-        self.stashed_model_static_state = _export_static_state(
-            self.tp_worker.worker.model_runner.model
-        )
-        self.memory_saver_adapter.pause()
-        self.flush_cache()
+        tags = recv_req.tags
+        import subprocess
+
+        if tags is None:
+            tags = ["weights", "kv_cache"]
+
+        if "kv_cache" in tags:
+            self.memory_saver_adapter.pause("kv_cache")
+            self.flush_cache()
+
+        if "weights" in tags:
+            self.stashed_model_static_state = _export_static_state(
+                self.tp_worker.worker.model_runner.model
+            )
+            self.memory_saver_adapter.pause("weights")
+
         return ReleaseMemoryOccupationReqOutput()
 
     def resume_memory_occupation(self, recv_req: ResumeMemoryOccupationReqInput):
-        self.memory_saver_adapter.check_validity(caller_name="resume_memory_occupation")
-        self.memory_saver_adapter.resume()
-        _import_static_state(
-            self.tp_worker.worker.model_runner.model, self.stashed_model_static_state
-        )
-        del self.stashed_model_static_state
+        tags = recv_req.tags
+        if tags is None or len(tags) == 0:
+            tags = ["weights", "kv_cache"]
+
+        if "weights" in tags:
+            self.memory_saver_adapter.resume("weights")
+            _import_static_state(
+                self.tp_worker.worker.model_runner.model,
+                self.stashed_model_static_state,
+            )
+            del self.stashed_model_static_state
+
+        if "kv_cache" in tags:
+            self.memory_saver_adapter.resume("kv_cache")
+            torch.cuda.synchronize()
+            time.sleep(3)
```

Comment:
nit: very quickly glanced and see this 3-second sleep, maybe it is a debug code that forgets to be removed (?)